test/arithmetic_test.rb

```diff
@@ -8,6 +8,36 @@ class Measured::ArithmeticTest < ActiveSupport::TestCase
     @four = Magic.new(4, :magic_missile)
   end
 
+  test 'can check for finite?' do
+    assert Magic.new(0, :magic_missile).finite?
+    refute Magic.new(Float::INFINITY, :magic_missile).finite?
+  end
+
+  test 'can check for infinite?' do
+    assert Magic.new(Float::INFINITY, :magic_missile).infinite?
+    refute Magic.new(0, :magic_missile).infinite?
+  end
+
+  test 'can check for zero?' do
+    assert Magic.new(0, :magic_missile).zero?
+    refute Magic.new(1, :magic_missile).zero?
+  end
+
+  test 'can check for nonzero?' do
+    assert Magic.new(10, :magic_missile), Magic.new(10, :magic_missile).nonzero?
```

Comment:
Small mistake here, probably a copy/paste error. But silently passes because the signature is `assert(condition, error_message)` and since the first is always true the second is ignored.

```suggestion
    assert Magic.new(10, :magic_missile).nonzero?
```

Reply:
Nice pickup, I blame using to much rspec :)